### DISCUSSION TITLE
File upload - S3 Metadata

### DISCUSSION THREAD

**Discussion by flashmediasolutions** [filamentphp/filament#12312] [Help]
### Package

Form builder

### Package Version

v3.2.65

### How can we help you?

When uploading a file to S3 bucket, I am unable to set metadata for the file on upload.
I need to supply the metadata to allow the file to be downloadable and not open in the browser.
In a perfect world ->downloadable() would auto supply the metadata on file upload.

![Screenshot 2024-04-12 at 10 51 21 AM](https://github.com/filamentphp/filament/assets/12239388/22eb5fe6-a86e-4659-a56c-c16c21f6749c)

Any insight would be greatly appreciated!

**Comment by flashmediasolutions** [comment]
Found this result for direct storage:

```php
Storage::disk('s3')->put($file_path, $file, [
     'ContentDisposition' => 'attachment'
]);
```

Still looking for the correct way to do with FileUpload.

**Reply by adrianlzx1996** [reply]
Are you able to find a solution?
